### ISSUE TITLE
[node-core-library] The public API contains `const enum`s.

### ISSUE THREAD

**Issue by StarFishing** [microsoft/rushstack#3058]
"[@rushstack/node-core-library] npm package file packaging error，Constants.js have empty export"

## Summary

I am introducing this enum value into the view
```
import {
  FileConstants,
} from '@rushstack/node-core-library';
```
when i run my code,i got error like:Error: Cannot read property 'PackageJson' of undefined
By looking at the npm package, I found that Constants.js package is empty

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| Package name: | @rushstack/node-core-library |
| Package version? |  3.43.2 |
| Operating system? |  Mac   |
| Would you consider contributing a PR? | No  |

**Comment by iclanton** [comment]
`FileConstants` is a `const enum`, meaning that it's only available to the TypeScript compiler at compile-time. I assume you're writing code in JavaScript or TypeScript with `isolatedModules` (or another tool that doesn't consider the typings information during transpilation).

https://github.com/microsoft/rushstack/pull/3060 updates the API to export conventional `enum`s.

**Comment by iclanton** [comment]
This was released with `@rushstack/node-core-library@3.44.0`.
